Classes/DataProcessing/GalleryProcessor.php

```diff
@@ -184,43 +198,15 @@ protected function prepareGalleryData()
                 $fileObj = $this->fileObjects[$fileKey] ?? null;
 
                 if ($fileObj) {
-                    $fileExtension = $this->processorConfiguration['fileExtension'] ?? null;
-
-                    if ($fileObj['properties']['type'] === 'image') {
-                        $image = $this->getImageService()->getImage((string)$fileObj['properties']['fileReferenceUid'], null, true);
-                        $fileObj = $this->getFileUtility()->processFile(
+                    if ((($fileObj['properties']['type'] ?? '') === 'image' || ($fileObj['type'] ?? '') === 'image')) {
+                        $src = $this->processorConfigurationObject->legacyReturn ? $fileObj['properties']['fileReferenceUid'] : $fileObj['fileReferenceUid'];
+                        $image = $this->getImageService()->getImage((string)$src, null, true);
+                        $fileObj = $this->getFileUtility()->process(
                             $image,
-                            array_merge(
-                                ['fileExtension' => $fileExtension],
```

Comment:
this change results into the images won´t be reused by ProcessedFileRepository::235, because the missing fileExtension results into another `configurationsha1` and on each request the image is processed